src/components/utils/test/posix_thread_test.cc

```diff
@@ -41,287 +43,466 @@ namespace utils_test {
+  EXPECT_EQ(NULL, thread->GetDelegate());
+  DeleteThread(thread);
 }
 
-TEST(PosixThreadTest,
-     CheckCreatedThreadNameChangeToShortName_ExpectThreadNameChangedToShort) {
+TEST_F(PosixThreadTest,
+       CheckCreatedThreadNameChangeToShortName_ExpectThreadNameChangedToShort) {
   // Arrange
   threads::Thread* thread = NULL;
   TestThreadDelegate* threadDelegate = new TestThreadDelegate();
-  AutoLock test_lock(test_mutex_);
-  // Create thread
-  ASSERT_NO_THROW(thread = CreateThread(threadName, threadDelegate));
-  // Start created thread
-  thread->start(threads::ThreadOptions(threads::Thread::kMinStackSize));
-  // Rename started thread. Name will be cut to 15 symbols + '\0'
-  // This is the limit in current POSIX thread implementation
-  thread->SetNameForId(thread->thread_handle(), test_thread_name);
-  // Name must be large enough to keep 16 symbols. Read previous comment
-  char name[MAX_SIZE];
-  int result = pthread_getname_np(thread->thread_handle(), name, sizeof(name));
-  if (!result) {
-    EXPECT_EQ(test_thread_name, std::string(name));
+
+  {
+    AutoLock test_lock(*threadDelegate);
+    // Create thread
+    ASSERT_NO_THROW(thread = CreateThread(kThreadName, threadDelegate));
+    // Start created thread
+    thread->Start(threads::ThreadOptions(threads::Thread::kMinStackSize));
+    // Rename started thread. Name will be cut to 15 symbols + '\0'
+    // This is the limit in current POSIX thread implementation
+    thread->SetNameForId(thread->ThreadHandle(), kTestThreadName);
+    // Name must be large enough to keep 16 symbols. Read previous comment
+    char name[kMaxSize];
+    int result = pthread_getname_np(thread->ThreadHandle(), name, sizeof(name));
+    if (!result) {
+      EXPECT_EQ(kTestThreadName, std::string(name));
+    }
+    threadDelegate->WaitFor(test_lock, 10000);
   }
-  cond_var_.WaitFor(test_lock, 10000);
+
   EXPECT_TRUE(threadDelegate->check_value());
-  DeleteThread(thread);
+
   delete threadDelegate;
-  EXPECT_EQ(NULL, thread->delegate());
+  EXPECT_EQ(NULL, thread->GetDelegate());
+  DeleteThread(thread);
 }
 
-TEST(PosixThreadTest, StartThread_ExpectThreadStarted) {
+TEST_F(PosixThreadTest, StartThread_ExpectThreadStarted) {
   // Arrange
   threads::Thread* thread = NULL;
   TestThreadDelegate* threadDelegate = new TestThreadDelegate();
-  AutoLock test_lock(test_mutex_);
-  // Create thread
-  ASSERT_NO_THROW(thread = CreateThread(threadName, threadDelegate));
-  // Start created thread
-  EXPECT_TRUE(
-      thread->start(threads::ThreadOptions(threads::Thread::kMinStackSize)));
-  cond_var_.WaitFor(test_lock, 10000);
+
+  {
+    AutoLock test_lock(*threadDelegate);
+    // Create thread
+    ASSERT_NO_THROW(thread = CreateThread(kThreadName, threadDelegate));
+    // Start created thread
+    EXPECT_TRUE(
+        thread->Start(threads::ThreadOptions(threads::Thread::kMinStackSize)));
+    threadDelegate->WaitFor(test_lock, 10000);
+  }
+
   EXPECT_TRUE(threadDelegate->check_value());
```

Comment:
rename check_value -> Value

Reply:
@LuxoftAKutsan 
Fixed in:https://github.com/LuxoftSDL/sdl_core/pull/50/commits/ca62978ee665d27ecfceffac8caf96ce740ecb51